library/oid.c

```diff
@@ -834,18 +834,52 @@ int mbedtls_oid_get_numeric_string(char *buf, size_t size,
     p = buf;
     n = size;
 
-    /* First byte contains first two dots */
-    if (oid->len > 0) {
-        ret = mbedtls_snprintf(p, n, "%d.%d", oid->p[0] / 40, oid->p[0] % 40);
-        OID_SAFE_SNPRINTF;
+    /* First subidentifier contains first two OID components */
+    i = 0;
+    value = 0;
+    if ((oid->p[0]) == 0x80) {
+        /* Overlong encoding is not allowed */
+        return MBEDTLS_ERR_OID_BUF_TOO_SMALL;
+    }
+
+    while (i < oid->len && ((oid->p[i] & 0x80) != 0)) {
+        /* Prevent overflow in value. */
+        if (((value << 7) >> 7) != value) {
```

Comment:
I think I've seen some of the ASN.1 PRs with a nicer way of doing this overflow check - can we use that for consistency?

Reply:
Ah yes, found it: https://github.com/Mbed-TLS/mbedtls/pull/6659/files#diff-22145eb62dd84e2b987fb2137eea71aea1b1cb8bead6089b1bfe24e61d3286deR848

```
-       if (((value << 7) >> 7) != value) {
+       if (value > (UINT_MAX >> 7)) {
            return MBEDTLS_ERR_OID_BUF_TOO_SMALL;
        }
```

Reply:
Ah okay, any examples you can point me to? Otherwise I'll have a search through the ASN1 code.